R/scale-discrete-.r

```diff
@@ -129,3 +128,26 @@ ScaleDiscretePosition <- ggproto("ScaleDiscretePosition", ScaleDiscrete,
     new
   }
 )
+
+# TODO: This is a clear candidate for vctrs once we adopt it
+new_mapped_discrete <- function(x) {
+  class(x) <- c("mapped_discrete", "numeric")
+  x
+}
+is_mapped_discrete <- function(x) inherits(x, "mapped_discrete")
+#' @export
+c.mapped_discrete <- function(..., recursive = FALSE) {
+  new_mapped_discrete(c(unlist(lapply(list(...), unclass))))
+}
+#' @export
+`[.mapped_discrete` <- function(x, ..., drop = TRUE) {
+  new_mapped_discrete(NextMethod())
+}
+#' @export
+`[<-.mapped_discrete` <- function(x, ..., value) {
```

Comment:
I don't think this is correct in general, but it should be fine for internal use. You might want to check that `value` is numeric just to be safe.

Reply:
this is mapped after `[<-.Date`, but I will (as mentioned) port this to vctrs once we embrace it